ocaml-lsp-server/src/semantic_highlighting.ml

```diff
@@ -0,0 +1,828 @@
+open Import
+open Fiber.O
+
+(* TODO:
+
+   - [x] support textDocument/semanticTokens/full
+
+   - [x] support textDocument/semanticTokens/full/delta
+
+   - [ ] support textDocument/semanticTokens/range *)
+
+module Token_type = struct
```

Comment:
Shouldn't this be in `Lsp.Types`?

Reply:
It could be but given the design requirements for token types, I'm not sure that's achievable in a desired manner.

- A client should be able to add their own token types
- Encoding of a token type (into an `int`) happens based on the position of the token type represented in the token type legend that the server sends to client

I'm not sure how to add token types to `Lsp.Types` such that they'd be useful

Reply:
> A client should be able to add their own token types

They can always do so by adding their own token type:

```
type my_token = Standard of Lsp.token | Mine of token
```

> Encoding of a token type (into an int) happens based on the position of the token type represented in the token type legend that the server sends to client

I suppose we don't need to add the encoding function then.

> I'm not sure how to add token types to Lsp.Types such that they'd be useful

Yes, it's only a little bit of value. Better than nothing I suppose.